core/src/test/java/org/camunda/bpm/engine/test/assertions/bpmn/ProcessInstanceAssertHasBusinessKeyTest.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
+ * under one or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership. Camunda licenses this file to you under the Apache License,
+ * Version 2.0; you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.camunda.bpm.engine.test.assertions.bpmn;
+
+import org.camunda.bpm.engine.runtime.ProcessInstance;
+import org.camunda.bpm.engine.test.Deployment;
+import org.camunda.bpm.engine.test.ProcessEngineRule;
+import org.camunda.bpm.engine.test.assertions.helpers.Failure;
+import org.camunda.bpm.engine.test.assertions.helpers.ProcessAssertTestCase;
+import org.junit.Rule;
+import org.junit.Test;
+
+import static org.camunda.bpm.engine.test.assertions.ProcessEngineTests.*;
+
+/**
+ * @author Dimitri Lavrenük (dimitri.lavr@gmail.com)
+ */
+public class ProcessInstanceAssertHasBusinessKeyTest extends ProcessAssertTestCase {
+
+  @Rule
+  public ProcessEngineRule processEngineRule = new ProcessEngineRule();
+
+  @Test
+  @Deployment(resources = {"bpmn/ProcessInstanceAssert-hasBusinessKey.bpmn"})
```

Comment:
I would propose to not create a new process model but reuse one of the existing, e.g. `ProcessInstanceAssert-isActive.bpmn`. As the process doesn't need any special design.

Reply:
I did this according to other tests that each have a separate process model definition file while most of them are exactly the same except for the process id. I found 25 of those files, some exmaples:
- ProcessInstanceAssert-isActive.bpmn
- ProcessInstanceAssert-isEnded.bpmn
- ProcessInstanceAssert-isStarted.bpmn
- ProcessInstanceAssert-hasVariables.bpmn
- ProcessInstanceAssert-hasNoVariables.bpmn

If I would replace it here with an existing file, the other tests should also be refactored, right?

Reply:
I guess both options are fine here, adding a new one is also OK if you already did it and it also separates the concerns well. You don't necessarily have to add one, but it's also fine to add one, I would say. 😃

Reply:
I agree with Tobi here.

Reply:
@yanavasileva does that mean that it is fine with the the `ProcessInstanceAssert-hasBusinessKey.bpmn` file or that existing test suits should be refactored to only use one single definition if it is the same?